src/web/nextui/src/app/eval/ResultsTable.tsx

```diff
@@ -138,6 +140,42 @@ function ResultsTable({
   const { evalId, table, setTable, config, inComparisonMode } = useMainStore();
   const { showToast } = useToast();
 
+  // Function to generate the URL with the row ID
+  const generateRowLink = useCallback((rowId: string) => {
+    const currentUrl = window.location.href.split('#')[0]; // Get the current URL without any hash
+    return `${currentUrl}#row-${rowId}`; // Append the row ID as a hash to the URL
+  }, []);
+
+  // Function to copy the link to the clipboard
+  const copyToClipboard = useCallback(
+    (text: string) => {
+      navigator.clipboard
+        .writeText(text)
+        .then(() => {
+          showToast('Link copied to clipboard', 'success');
+        })
+        .catch((err) => {
+          console.error('Could not copy text: ', err);
+          showToast('Failed to copy link', 'error');
+        });
+    },
+    [showToast],
+  );
+
+  // Scroll to the row when the URL contains a hash
+  useEffect(() => {
```

Comment:
Two things I'm noticing here:
- We need to account for pagination - currently navigating to a link on page 2 (more than 50 results) does not go to it
- The results page supports other query parameters such as search (which is used in the red team report view).  So it would be good to have more robust parsing, e.g. for a hash like `row-id=foo&search=abc123` (this also applies to generateRowLink)